jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmappers/FailureExceptionMapper.java

```diff
@@ -0,0 +1,70 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+
+import com.cerner.beadledom.jaxrs.models.JsonError;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.spi.Failure;
+import org.jboss.resteasy.spi.ReaderException;
+import org.jboss.resteasy.spi.WriterException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Exception Mapper for all unhandled Exceptions extending from the {@link Failure} hierarchy
+ * (specifically for Resteasy Built-in Internally-Thrown Exceptions like {@link ReaderException},
+ * {@link WriterException}, etc.) thrown by the service. See
+ * http://docs.jboss.org/resteasy/docs/3.1.4.Final/userguide/html_single/index.html#builtinException
+ * for more information.
+ *
+ * @author Cal Fisher
+ */
+@Provider
+public class FailureExceptionMapper implements ExceptionMapper<Failure> {
+
+  private static final Logger logger = LoggerFactory.getLogger(FailureExceptionMapper.class);
+
+  /**
+   * Generates a Response object with a Status of the Failure, Content-Type 'application/json', and
+   * a JsonError entity containing details about the unhandled exception in Json format.
+   *
+   * @param exception the Failure exception that was not handled
+   * @return a json response with the exception's status or 500 if the exception has no response
+   */
+  @Override
+  public Response toResponse(Failure exception) {
+
+    logger.error("An unhandled Failure exception was thrown.", exception);
+
+    int statusCode;
+    if (exception.getResponse() != null) {
+      statusCode = exception.getResponse().getStatus();
+    } else {
+      statusCode = INTERNAL_SERVER_ERROR.getStatusCode();
+    }
+
+    if (statusCode >= 400 && statusCode < 500) {
```

Comment:
Since you are already logging the exception on line 42, this can be removed.

Reply:
It seems like this was wanted so client errors (400-499) are logged as warnings while server errors (500-599) are logged as errors.

I've heard before that some 404 exceptions that are getting logged are not desired at all. Does this same sentiment apply to all 400-level errors or is there value in logging out all 400-level errors (at least in this case of a Failure exception)?

A 400 error could occur here if the `MessageBodyReader` fails with an unknown exception or the XML marshaller (which likely wouldn't affect JSON services) throws an unknown exception.

Reply:
Another thought. What might be good to do (but probably as a separate issue/PR) would be to allow configuring a list of status codes to not be logged.

Reply:
@johnlcox That would be an interesting idea. I'm sure some teams would prefer the option to customize which errors get logged and which do not. If it becomes an issue in the future, we should consider that option, but I think for right now it might not be a bad idea to leave the logging in here. (The previous `PlainTextWebApplicationExceptionMapper` was logging an error every time, so a warning for non-500+ errors would be an improvement nonetheless.)

Regardless: I removed the extra logging code that was left in accidentally: https://github.com/cerner/beadledom/pull/45/commits/865a017a55ce93a2c50c539260b17f6bbcdfe110#diff-94f0fc074c60a4a4d74f16bebf75afc9L42